### PR TITLE
merge same analyzers in different location added multiple times to a …

### PR DESCRIPTION
**Pull request by heejaechang**
…project to one.

fixes this - https://devdiv.visualstudio.com/DevDiv/_workitems?id=479136&src=WorkItemMention&src-action=artifact_link&fullScreen=true&_a=edit

### CODE REVIEW

**Comment by heejaechang** [comment]
@dotnet/roslyn-analysis @mavasani can you take a look?

**Comment by heejaechang** [comment]
I think this scenario is corner case. so I am just making VS not to crash. we can think more expansive way to fix this if we really want to support this scenario.

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure 
Roslyn.VisualStudio.IntegrationTests.Basic.BasicEncapsulateField.EncapsulateThroughLightbulbDefinitionsOnly [FAILED]
https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_vs-integration_prtest/7350/

**Comment by heejaechang** [comment]
retest windows_release_vs-integration_prtest  please

**Comment by heejaechang** [comment]
@mavasani can you take a look?

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs [hunk 1]
```diff
@@ -215,10 +219,13 @@ private void ReportAnalyzerExceptions(Project project, ImmutableDictionary<Diagn
...
 
-            private Dictionary<string, DiagnosticAnalyzer> CreateAnalyzerMap(IEnumerable<DiagnosticAnalyzer> analyzers)
+            private void AppendAnalyzerMap(IEnumerable<DiagnosticAnalyzer> analyzers, Dictionary<string, DiagnosticAnalyzer> analyzerMap)
             {
-                // TODO: this needs to be cached. we can have 300+ analyzers
-                return analyzers.ToDictionary(a => a.GetAnalyzerIdAndVersion().Item1, a => a);
+                foreach (var analyzer in analyzers)
+                {
+                    // user might have included exact same analyzer twice as project analyzers explicitly. we consider them as one
+                    analyzerMap[analyzer.GetAnalyzerId()] = analyzer;
```
Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs]: `GetAnalyzerIdAndVersion` was replaced with a call to `GetAnalyzerId`. Was this intentional?

Comment (heejaechang) [reply src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs]: http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/AnalyzerHelper.cs,132

GetAnalyzerIdAndVersion().Item1 == analyzer.GetAnalyzerId()

**Pull request merged**